web/packages/teleterm/src/services/tshd/createClient.ts

```diff
@@ -119,71 +88,40 @@ export function createTshdClient(
       startKey,
       limit,
     }: types.GetResourcesParams) {
-      const req = api.GetAppsRequest.create({
+      const { response } = await tshd.getApps({
         clusterUri,
         searchAsRoles,
         startKey,
         search,
         query,
         limit,
+        sortBy: sort ? `${sort.fieldName}:${sort.dir.toLowerCase()}` : '',
       });
 
-      if (sort) {
-        req.sortBy = `${sort.fieldName}:${sort.dir.toLowerCase()}`;
-      }
-
-      return new Promise<types.GetAppsResponse>((resolve, reject) => {
-        tshd.getApps(req, (err, response) => {
-          if (err) {
-            reject(err);
-          } else {
-            resolve(response as types.GetAppsResponse);
-          }
-        });
-      });
+      return response as types.GetAppsResponse;
     },
 
     async listGateways() {
-      const req = api.ListGatewaysRequest.create();
-      return new Promise<types.Gateway[]>((resolve, reject) => {
-        tshd.listGateways(req, (err, response) => {
-          if (err) {
-            reject(err);
-          } else {
-            resolve(response.gateways as types.Gateway[]);
-          }
-        });
-      });
+      const { response } = await tshd.listGateways({});
+
+      return response.gateways as types.Gateway[];
     },
 
     async listLeafClusters(clusterUri: uri.RootClusterUri) {
-      const req = api.ListLeafClustersRequest.create({ clusterUri });
-      return new Promise<types.Cluster[]>((resolve, reject) => {
-        tshd.listLeafClusters(req, (err, response) => {
-          if (err) {
-            reject(err);
-          } else {
-            resolve(response.clusters as types.Cluster[]);
-          }
-        });
-      });
+      const { response } = await tshd.listLeafClusters({ clusterUri });
+
+      return response.clusters as types.Cluster[];
     },
 
-    async listRootClusters(abortSignal?: types.TshAbortSignal) {
-      return withAbort(abortSignal, callRef => {
-        return new Promise<types.Cluster[]>((resolve, reject) => {
-          callRef.current = tshd.listRootClusters(
-            api.ListClustersRequest.create(),
-            (err, response) => {
-              if (err) {
-                reject(err);
-              } else {
-                resolve(response.clusters as types.Cluster[]);
-              }
-            }
-          );
-        });
-      });
+    async listRootClusters(abortSignal?: CloneableAbortSignal) {
+      const { response } = await tshd.listRootClusters(
+        {},
```

Comment:
Having to pass an empty object is unfortunate, but I already ran into the same problem when working on tshd events.